src/median_of_median.c

```diff
@@ -32,6 +32,48 @@ int quick_select(int A[], int n, int k){
   else return quick_select(A+1, j-1, k);
 }
 
+int m(int A[],int n){//median_of_medianのpivotを求める補助関数
+    if (n > 5) {
+        int X[n/5];
+        int i;
+        for(i = 0;i < (n/5)-1;i++){
+            int Z[5] = {A[5*i],A[5*i+1],A[5*i+2],A[5*i+3],A[5*i+4]};
+            X[i] = quick_select(Z,5,3);//中央値をquick_selectを用いて求める(他にいい方法があれば知りたいです)
```

Comment:
quick_select(A,n,k)は配列の`k+1`番目の値を返す関数です．長さ５の配列の中央値は4番目ではありません．
コメントに書いてある質問とは関係ないですが下のように書くとメモリが節約できます．
```suggestion
            X[i] = quick_select(X+5i,5,2);
```